client/js/widgets/widget.js

```diff
@@ -503,149 +503,11 @@ export class Widget extends StateManaged {
-          case 'random':
-          case 'round':
-          case 'sign':
-          case 'sqrt':
-          case 'trunc':
-            v = Math[o](x);
-            break;
-          case 'E':
-          case 'LN2':
-          case 'LN10':
-          case 'LOG2E':
-          case 'LOG10E':
-          case 'PI':
-          case 'SQRT1_2':
-          case 'SQRT2':
-            v = Math[o];
-            break;
-
-          // String operations
-          case 'length':
-            v = x.length;
-            break;
-          case 'parseFloat':
-            v = parseFloat(x);
-            break;
-          case 'toLowerCase':
-          case 'toUpperCase':
-          case 'trim':
-          case 'trimStart':
-          case 'trimEnd':
-            v = x[o]();
-            break;
-          case 'charAt':
-          case 'charCodeAt':
-          case 'codePointAt':
-          case 'concat':
-          case 'includes':
-          case 'endsWith':
-          case 'indexOf':
-          case 'lastIndexOf':
-          case 'localeCompare':
-          case 'match':
-          case 'padEnd':
-          case 'padStart':
-          case 'repeat':
-          case 'search':
-          case 'split':
-          case 'startsWith':
-          case 'toFixed':
-          case 'toLocaleLowerCase':
-          case 'toLocaleUpperCase':
-            v = x[o](y);
-            break;
-          case 'replace':
-          case 'replaceAll':
-          case 'substr':
-            v = x[o](y, z);
-            break;
-
-          // Array operations
-          // 'length' should work the same as for strings
-          case 'getIndex':
-            v = x[y];
-            break;
-          case 'setIndex':
-            v[x] = y;
-            break;
-          case 'from':
-          case 'isArray':
-            v = Array[o](x);
-            break;
-          case 'concatArray':
-            v = x.concat(y);
-            break;
-          case 'pop':
-          case 'reverse':
-          case 'shift':
-          case 'sort':
-            v = x[o]();
-            break;
-          case 'includes':
-          case 'indexOf':
-          case 'join':
-          case 'lastIndexOf':
-            v = x[o](y);
-            break;
-          case 'slice':
-            v = x[o](y, z);
-            break;
-          case 'push':
-          case 'unshift':
-            v[o](x);
-            break;
-
-          // random values
-          case 'randInt':
-            v = Math.floor((Math.random() * (y - x + 1)) + x);
-            break;
-          case 'randRange':
-            v = Math.round(Math.floor((Math.random() * (y - x) / (z || 1))) * (z || 1) + x);
-            break;
-          default:
-            v = null;
+          if (compute_ops.find(op => op.name == o) !== undefined) {
+            v = compute_ops.find(op => op.name == o).call(v, x, y, z);
+          }else {
             problems.push(`Operation ${o} is unsupported.`);
-            return v;
+            return v = null;
```

Comment:
This changes behavior. Or am I missing something?

Reply:
If this helps, I ran a fake operation on both servers, and both times the variable was set to null.

![image](https://user-images.githubusercontent.com/78324099/119609726-afc00e00-bdbd-11eb-9cc5-88a3cb73e0b1.png)

Reply:
It should be the same behavior. The original code was:
```
          default:
            v = null;
            problems.push(`Operation ${o} is unsupported.`);
            return v;
```

Reply:
My bad.